charms/worker/k8s/src/charm.py

```diff
@@ -890,17 +888,16 @@ def _announce_certificates_config(self) -> None:
         if not (provider := self.config.get(CONFIG_BOOTSTRAP_CERTIFICATES)):
             raise ReconcilerError("Missing certificates provider")
 
-        relation = self.model.get_relation(CLUSTER_WORKER_RELATION)
-        if not relation:
+        for rel in self.model.relations[CLUSTER_WORKER_RELATION]:
+            rel.data[self.app][CLUSTER_CERTIFICATES_KEY] = str(provider)
+            kubelet_formatter = str(self.config.get(KUBELET_CN_FORMATTER_CONFIG_KEY))
+            rel.data[self.app][CLUSTER_CERTIFICATES_KUBELET_FORMATTER_KEY] = kubelet_formatter
+            domain_name = str(self.config.get(COMMON_NAME_CONFIG_KEY))
+            rel.data[self.app][CLUSTER_CERTIFICATES_DOMAIN_NAME_KEY] = domain_name
+        else:
```

Comment:
@mateoflorido i spotted this one when looking for other issues where this could be broken.